.github/workflows/capgen_unit_tests.yaml

```diff
@@ -0,0 +1,22 @@
+name: Capgen Unit Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      #Trigger workflow on push to any branch or branch heirarchy:
+      - '**'
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: update repos and install dependencies
+      run: sudo apt-get update && sudo apt-get install -y build-essential gfortran cmake python3 git
+    - name: create python venv
+      run: python3 -m venv ./ccpp-env && source ./ccpp-env/bin/activate && pip3 install -r requirements.txt
```

Comment:
Is this step actually needed?  It looks like the `requirements.txt` file is bringing in packages that I don't think are actually being used by the tests in `run_tests.sh` (although I could be wrong here).

Reply:
Good catch!  I just saw the python scripts in the `test` directory and just assumed they were needed but they are indeed not used.  Removed.